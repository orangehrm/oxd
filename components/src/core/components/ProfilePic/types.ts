export const SIZE_EXRTA_SMALL = 'small';
export const SIZE_SMALL = 'small';
export const SIZE_MEDIUM = 'medium';

export const SIZES = [SIZE_EXRTA_SMALL, SIZE_SMALL, SIZE_MEDIUM];

export type ImageSize = typeof SIZE_EXRTA_SMALL | typeof SIZE_SMALL | typeof SIZE_MEDIUM;

export const TARGET_SELF = '_self';
export const TARGET_BLANK = '_blank';
export const TARGET_PARENT = '_parent';
export const TARGET_TOP = '_top';

export const TARGETS = [TARGET_SELF, TARGET_BLANK, TARGET_PARENT, TARGET_TOP];

export type TargetTypes =
  | typeof TARGET_SELF
  | typeof TARGET_BLANK
  | typeof TARGET_PARENT
  | typeof TARGET_TOP;
