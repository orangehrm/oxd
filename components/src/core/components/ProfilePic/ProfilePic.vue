<template>
  <img class="header-image" :src="imageUrl" />
</template>

<script lang="ts">
import {defineComponent} from 'vue';
export default defineComponent({
  name: 'ProfilePic',
  props: {
    imageUrl: {
      type: String,
      required: true,
    },
  },
});
</script>
<style lang="scss" scoped>
.header-image {
  border-radius: 50%;
  width: 3.75rem;
  height: auto;
  max-height: 3.75rem;
  cursor: pointer;
}
</style>
