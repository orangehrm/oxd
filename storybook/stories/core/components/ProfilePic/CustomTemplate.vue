<template>
  <oxd-profile-pix>
    <img :src="data.imageSrc" />
  </oxd-profile-pix>
</template>

<script>
import ProfilePic from '@orangehrm/oxd/core/components/ProfilePic/ProfilePic';
export default {
  components: {
    'oxd-profile-pix': ProfilePic,
  },
  data() {
    return {
      data: {
        imageSrc: require('@orangehrm/oxd/assets/images/user-default-400.png'),
      },
    };
  },
};
</script>
