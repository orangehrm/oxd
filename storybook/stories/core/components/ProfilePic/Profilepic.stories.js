import ProfilePic from '@orangehrm/oxd/core/components/ProfilePic/ProfilePic';

export default {
  title: 'Example/ProfilePic',
  component: ProfilePic,
};

const Template = args => ({
  setup() {
    return {args};
  },
  components: {'oxd-profile-pic': ProfilePic},
  template: '<oxd-profile-pic v-bind="args" />',
});

export const Image = Template.bind({});
const defaultPic = require('@orangehrm/oxd/assets/images/user-default-400.png');
Image.args = {
  imageUrl: defaultPic,
};
