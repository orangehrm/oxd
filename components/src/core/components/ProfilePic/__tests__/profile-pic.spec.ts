import {mount} from '@vue/test-utils';
import ProfilePic from '@orangehrm/oxd/core/components/ProfilePic/ProfilePic.vue';

describe('ProfilePic.vue', () => {
  it('should renders OXD ProfilePic for image URL', () => {
    const wrapper = mount(ProfilePic, {
      props: {imageUrl: '../../assests/images/user-default-400.png'},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
