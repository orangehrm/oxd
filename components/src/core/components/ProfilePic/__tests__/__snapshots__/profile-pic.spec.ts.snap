// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePic.vue should renders OXD ProfilePic for image URL 1`] = `<img class="header-image" src="../../assests/images/user-default-400.png">`;
